src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -87,9 +89,25 @@ internal class FhirCosmosSearchService : SearchService
                 });
         }
 
-        public override Task<IReadOnlyList<(short ResourceTypeId, string Name)>> GetUsedResourceTypes(CancellationToken cancellationToken)
+        public override async Task<IReadOnlyList<string>> GetUsedResourceTypes(CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            QueryDefinition sqlQuerySpec = new QueryDefinition(@"SELECT DISTINCT VALUE r.resourceTypeName
+                FROM root r
+                WHERE r.isSystem = false
+                ORDER BY r.resourceTypeName");
```

Comment:
Do we know how this will perform on large databases? Something I'd like to find out

Reply:
I think if you remove the `ORDER BY` this will perform fine.

I'm testing (*adopting* 😄 ) this function for parallel Gen 1 export. I have a 500GB collection with 50K RU. The orig query above appears to be scanned record by record and the query times out for me for subsequent results after the first.

Removing the `ORDER BY` results in near instant execution. I can get more precise info if needed.

Reply:
Mikael and I talked about this. We have the issue where the request will fail to return a continuation token if we don't include an `ORDER BY` with a `DISTINCT`. I'm still looking into solutions.